src/main/java/com/google/codeu/servlets/LikeServlet.java

```diff
@@ -0,0 +1,82 @@
+package com.google.codeu.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.codeu.data.Datastore;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.servlet.ServletContext;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+import com.google.codeu.data.User;
+
+/** Handles fetching and saving likes and not likes. */
+@WebServlet("/like")
+public class LikeServlet extends HttpServlet {
+
+    private Datastore datastore;
+
+    @Override
+    public void init() {
+        datastore = new Datastore();
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        System.out.println("Posting like in servlet...");
+
+        UserService userService = UserServiceFactory.getUserService();
+
+        String userEmail = userService.getCurrentUser().getEmail();
+
+        User user = datastore.getUser(userEmail);
+        // Note: sometimes the current user is not stored in the datastore!
+        if (user == null) {
+            System.out.println("The current user is not in Datastore. Adding them...");
+            user = new User(userEmail, null, new HashSet<String>(), new HashSet<String>());
+        }
+
+        // Get who the user likes
+        String prospectiveEmail = request.getParameter("email");
+        System.out.println("prospective email: " + prospectiveEmail);
+        User prospectiveMatch = datastore.getUser(prospectiveEmail);
+        // Note: sometimes the prospective user is not stored in the datastore!
+        if (prospectiveMatch == null) {
```

Comment:
It the prospectiveMatch was not stored in the datastore previously, how did the user like it?